sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/CheckpointStore/BlobsCheckpointStoreTests.cs

```diff
@@ -433,6 +441,241 @@ public async Task ListCheckpointsConsidersDataInvalidWithNoOffsetOrSequenceNumbe
             mockLogger.Verify(log => log.InvalidCheckpointFound(partitionId, FullyQualifiedNamespace, EventHubName, ConsumerGroup));
         }
 
+        /// <summary>
+        ///   Verifies basic functionality of ListCheckpointsAsync and ensures the starting position is set correctly.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ListCheckpointsPreferredNewCheckpointOverLegacy()
+        {
+            string partitionId = Guid.NewGuid().ToString();
+            var blobList = new List<BlobItem>{
+
+                BlobsModelFactory.BlobItem($"{FullyQualifiedNamespace}/{EventHubName}/{ConsumerGroup}/checkpoint/{partitionId}",
+                                            false,
+                                            BlobsModelFactory.BlobItemProperties(true, lastModified: DateTime.UtcNow, eTag: new ETag(MatchingEtag), contentLength: 0),
+                                            "snapshot",
+                                            new Dictionary<string, string>
+                                            {
+                                                {BlobMetadataKey.OwnerIdentifier, Guid.NewGuid().ToString()},
+                                                {BlobMetadataKey.SequenceNumber, "960182"},
+                                                {BlobMetadataKey.Offset, "14"}
+                                            }),
+                BlobsModelFactory.BlobItem($"{FullyQualifiedNamespace}/{EventHubName}/{ConsumerGroup}/{partitionId}",
+                                           false,
+                                           BlobsModelFactory.BlobItemProperties(true, lastModified: DateTime.UtcNow, eTag: new ETag(MatchingEtag)),
+                                           "snapshot")
+            };
+
+            var containerClient = new MockBlobContainerClient() { Blobs = blobList };
+            containerClient.AddBlobClient($"{FullyQualifiedNamespace}/{EventHubName}/{ConsumerGroup}/{partitionId}", client =>
+            {
+                client.Content = Encoding.UTF8.GetBytes("{" +
+                                                            "\"PartitionId\":\"0\"," +
+                                                            "\"Owner\":\"681d365b-de1b-4288-9733-76294e17daf0\"," +
+                                                            "\"Token\":\"2d0c4276-827d-4ca4-a345-729caeca3b82\"," +
+                                                            "\"Epoch\":386," +
+                                                            "\"Offset\":\"13\"," +
+                                                            "\"SequenceNumber\":960180}");
```

Comment:
(extreme) nit:  If the opening curly is on its own line, shouldn't the closing one follow that convention as well?

Reply:
Leaving one comment here but seems to apply to each of the JSON checkpoints.